File: fstar/_output/ocaml/Space_Types.ml
open Prims
type cell = FStar_UInt64.t
type universe_id = Prims.nat
type universe_name = Prims.string
type discipline =
  | Unrestricted 
  | Affine 
  | Linear 
let uu___is_Unrestricted (projectee : discipline) : Prims.bool=
  match projectee with | Unrestricted -> true | uu___ -> false
let uu___is_Affine (projectee : discipline) : Prims.bool=
  match projectee with | Affine -> true | uu___ -> false
let uu___is_Linear (projectee : discipline) : Prims.bool=
  match projectee with | Linear -> true | uu___ -> false

File: fstar/_output/fsharp/Space_Types.fs
#light "off"
module Space_Types

type cell =
FStar_UInt64.t


type universe_id =
Prims.nat


type universe_name =
Prims.string

type discipline =
| Unrestricted
| Affine
| Linear


let uu___is_Unrestricted : discipline  ->  Prims.bool = (fun ( projectee  :  discipline ) -> (match (projectee) with
| Unrestricted -> begin
true
end
| uu___ -> begin
false
end))


let uu___is_Affine : discipline  ->  Prims.bool = (fun ( projectee  :  discipline ) -> (match (projectee) with
| Affine -> begin
true
end
| uu___ -> begin
false
end))


let uu___is_Linear : discipline  ->  Prims.bool = (fun ( projectee  :  discipline ) -> (match (projectee) with
| Linear -> begin
true
end
| uu___ -> begin
false
end))




